sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -38,6 +37,9 @@ import (
 
 // TODO(herohde) 2/8/2017: for now, assume we stage a full binary (not a plugin).
 
+// ProfileWriter is set by runners to emit profiles generated by the harness.
+var ProfileWriter = func(string, io.Reader) error { return nil }
```

Comment:
How are runners supposed to set them? I don't see it being set at all.

Reply:
The exported variables can be set by the runner.

Let me add the local-file writing implementation back in for the direct runner, so that functionality exists, and it's documentation by example.

Should I plumb these types through the reflector so these variables exist in the beam namespace, or is it OK for the runner to just import this package directly to set them?

Reply:
I meant: how are runners supposed to set them correctly? To be present at runtime, it would have to be done at init time -- before a runner is even selected. Multiple runners might overwrite each other. I think we need some kind of registration and enablement though pipeline options set by the selected runner.

No need to forward, unless the user is supposed to interact with it.

Reply:
I see your point. I've been thinking about one monolithic runner, and wasn't considering that beamx allows for a runtime selection of the runner.

Let me try out some ideas and I'll have a commit to discuss the registration mechanism.